src/bin/genemichaels.rs

```diff
@@ -199,7 +237,104 @@ fn main() {
             Offable::On(_) => true,
         },
     };
-    if args.files.is_empty() {
+    if args.workspace || args.folder {
```

Comment:
There's a bunch of branches checking args.workspace/args.folder here.  I think it would be cleaner to split this into two separate branches entirely, and then write a function for any shared functionality.

Specifically here I think the main shared functionality would be walking a directory and formatting in parallel, which shouldn't have too many external dependencies to pass in.

Reply:
Good call.